pipenv/core.py

```diff
@@ -1335,23 +1339,30 @@ def pip_install(
 
     # Try installing for each source in project.sources.
     if index:
-        if not is_valid_url(index):
-            index = project.find_source(index).get("url")
-        sources = [{"url": index}]
+        try:
+            index_source = project.find_source(index)
+            index_source = index_source.copy()
+        except SourceNotFound:
+            src_name = project.src_name_from_url(index)
+            verify_ssl = True if index not in trusted_hosts else False
```

Comment:
You may have considered this already, but while I was testing the issue `index` was set to something like `https://MY_INDEX:8080`, whereas `trusted_hosts` was just a list like`['MY_INDEX']`.

Reply:
no I absolutely in no way was thinking about that, I should probably parse netlocs :D thanks for the comment!

Reply:
https://github.com/pypa/pipenv/pull/2993/commits/272e98cb6749089341dd05e509a283d358cf2a59